tests/Model/IndexInputTest.php

```diff
@@ -40,16 +42,22 @@ public function testConstructorShouldRequireNameToBeString(): void
         new IndexInput(['key' => ['x' => 1], 'name' => 1]);
     }
 
-    /** @dataProvider provideExpectedNameAndKey */
-    public function testNameGeneration($expectedName, array $key): void
+    /**
+     * @dataProvider provideExpectedNameAndKey
+     * @param array|object $key
+     */
+    public function testNameGeneration($expectedName, $key): void
```

Comment:
```suggestion
    public function testNameGeneration(string $expectedName, $key): void
```

Might as well add this while we're changing this line.